graphql-dgs/src/main/kotlin/com/netflix/graphql/dgs/internal/ArgumentResolver.kt

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2022 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.graphql.dgs.internal
+
+import graphql.schema.DataFetchingEnvironment
+import org.springframework.core.MethodParameter
+import org.springframework.core.Ordered
+import org.springframework.core.annotation.Order
+
+@Order(Ordered.HIGHEST_PRECEDENCE)
+interface ArgumentResolver {
```

Comment:
Docs.

* Why I should be interested in an `ArgumentResolver`
* When should I provide my own `ArgumentResolver`
* Are there any implementation details I need to be aware in case I provide one? e.g. when should I change the `@Order`.